api/s2n.h

```diff
@@ -436,6 +436,23 @@ extern int s2n_connection_is_session_resumed(struct s2n_connection *conn);
 S2N_API
 extern int s2n_connection_is_ocsp_stapled(struct s2n_connection *conn);
 
+typedef enum {
+    S2N_HANDSHAKE_SIGNATURE_RSA,
+    S2N_HANDSHAKE_SIGNATURE_ECDSA,
+    S2N_HANDSHAKE_SIGNATURE_RSA_PSS_RSAE,
+    S2N_HANDSHAKE_SIGNATURE_RSA_PSS_PSS
```

Comment:
I don't have enough knowledge to really say, I'm not familiar with this aspect, my educated guess is that it seems we can collapse those two into a single `RSA_PSS` enum, since the signer will be aware of the certificate type. That is from the perspective of my use case, I'm not sure if that falls apart elsewhere.

If this sounds good to you, I'll make this change.

Reply:
Eh no I think you were right. They're different "SignatureSchemes" in TLS, even controlling for hash algorithm: https://tools.ietf.org/rfc/rfc8446#section-4.2.3 So we should probably be explicit.